files/appuio-pruner-job.yml

```diff
@@ -0,0 +1,66 @@
+---
+kind: Template
+apiVersion: v1
+metadata:
+  name: appuio-pruner-job
+  annotations:
+    description: Install an APPUiO Pruner Job
+    version: '2.0.0'
+
+parameters:
+  - description: Kind of objects that are pruned, eg "images"
+    name: KIND
+    required: true
+
+  - description: Image to use for the pruner jobs
+    name: IMAGE
+    required: true
+
+  - description: Schedule to use for the cronjobs
+    name: SCHEDULE
+    value: "@hourly"
+
+  - description: Command to execute
+    name: COMMAND
+    required: true
+
+objects:
+
+  - apiVersion: batch/v1beta1
+    kind: CronJob
+    metadata:
+      name: prune-${KIND}
+      labels:
+        app: appuio-pruner
+    spec:
+      schedule: ${SCHEDULE}
+      failedJobsHistoryLimit: 3
+      successfulJobsHistoryLimit: 1
+      concurrencyPolicy: Forbid
+      startingDeadlineSeconds: 86400
+      jobTemplate:
+        spec:
+          activeDeadlineSeconds: 7200
+          completions: 1
+          # Don't retry failed executions
+          backoffLimit: 0
+          template:
+            metadata:
+              labels:
+                name: prune-${KIND}
+                app: appuio-pruner
+            spec:
+              containers:
+                - name: pruner
+                  image: ${IMAGE}
+                  imagePullPolicy: IfNotPresent
+                  args: ${{COMMAND}}
+                  resources:
+                    requests:
+                      cpu: 100m
+                      memory: 128Mi
+                    limits:
+                      cpu: 250m
+                      memory: 1Gi
+              serviceAccountName: pruner
+              restartPolicy: Never
```

Comment:
I was under the impression to have commented on this before, but I can't find it anymore: We must support OpenShift 3.7 still, but this cron job definition will make the cluster unhappy. Can you provide backport for 3.7 like we with the Gluster recycler?

Reply:
IMHO we do not need to support 3.7 anymore, we can just let the previous pruner do it's job there until we upgrade.

Reply:
Unfortunately the pruner is deployed automatically, so we'd at least need code to not apply it on 3.7 clusters anymore (not part of this role, but an internal playbook).

Reply:
Yup, as per README v2 of the pruner only supports 3.9 and on. Older clusters must use ~v1.0.9. I'll make sure the post-config playbook only includes this role on 3.9 and on.